src/Http/RequestFactory.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http;
+
+use Tempest\Support\ArrayHelper;
+
+final readonly class RequestFactory
```

Comment:
This is just a quick PoC. Yes… we could refactor this to use PSR

Reply:
Huh! That's a thought. 🤔 🙃